src/matching/quoteMatcher.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * QuoteMatcher matches quoted strings, respecting escaped quotes (\") and frieds
+ */
+export class QuoteMatcher {
+  static escapeChar = "\\";
+
+  static findOpening(char: string, corpus: string, start: number): number {
+    // first, search backwards to see if we could be inside a quote
```

Comment:
And yes, it doesn't handle the case of:

```
"foo" <cursor> "bar"
```

where a `ci"` will leave you with

```
"foo"<cursor>"bar"
```

But, this is also how vim treats this.